From: blogic <blogic@3c298f89-4303-0410-b956-a3cf2f4a3e73>
Date: Tue, 7 May 2013 12:35:07 +0000
Subject: grub2: Add sub package grub-editenv for target installation

grub-editenv allows to modify grub2 environment files. Add a new package
that build grub2 for the target and packs up grub-editenv.

Signed-off-by: Helmut Schaa <helmut.schaa@googlemail.com>

diff --git a/package/grub2/Makefile b/package/grub2/Makefile
index b606f54..dd490ff 100644
--- a/package/grub2/Makefile
+++ b/package/grub2/Makefile
@@ -16,7 +16,6 @@ PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=@GNU/grub
 PKG_MD5SUM:=e927540b6eda8b024fb0391eeaa4091c
 
-PKG_HOST_ONLY:=1
 HOST_BUILD_PARALLEL:=1
 PKG_BUILD_DEPENDS:=grub2/host
 
@@ -32,6 +31,23 @@ define Package/grub2
   DEPENDS:=@TARGET_x86
 endef
 
+define Package/grub2-editenv
+  CATEGORY:=Utilities
+  SECTION:=utils
+  TITLE:=Grub2 Environment editor
+  URL:=http://www.gnu.org/software/grub/
+  DEPENDS:=@TARGET_x86
+endef
+
+define Package/grub2-editenv/description
+	Edit grub2 environment files.
+endef
+
+CONFIGURE_ARGS += \
+	--target=$(REAL_GNU_TARGET_NAME) \
+	--disable-werror \
+	--disable-nls
+
 HOST_CONFIGURE_ARGS += \
 	--target=$(REAL_GNU_TARGET_NAME) \
 	--sbindir="$(STAGING_DIR_HOST)/bin" \
@@ -47,5 +63,11 @@ define Host/Configure
 	$(Host/Configure/Default)
 endef
 
+define Package/grub2-editenv/install
+	$(INSTALL_DIR) $(1)/usr/sbin
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/grub-editenv $(1)/usr/sbin/
+endef
+
 $(eval $(call HostBuild))
 $(eval $(call BuildPackage,grub2))
+$(eval $(call BuildPackage,grub2-editenv))
